org.opentosca.planbuilder.integration/src/org/opentosca/planbuilder/integration/layer/AbstractImporter.java

```diff
@@ -59,9 +76,16 @@
      * @return a List of Plans
      */
     public List<AbstractPlan> buildPlans(final AbstractDefinitions defs, final String csarName) {
+                
         final List<AbstractPlan> plans = new ArrayList<>();
 
-        final AbstractSimplePlanBuilder buildPlanBuilder = new BPELBuildProcessBuilder();
+        AbstractSimplePlanBuilder buildPlanBuilder = new BPELBuildProcessBuilder();
+
+        if (!new BPELSituationAwareBuildProcessBuilder().buildPlans(csarName, defs).isEmpty()) {
+
+            buildPlanBuilder = new BPELSituationAwareBuildProcessBuilder();
```

Comment:
does the instance become unusable or why create the builder twice?

Reply:
Changed